### PR TITLE
Allow spawn to be called with retries

### PR DESCRIPTION
**Pull request by tejohnso**
@fjvallarino @ahmedalsudani please review

### CODE REVIEW

File: platform.js [hunk 1]
```diff
@@ -102,8 +102,9 @@ module.exports = {
       detached: true
     }).unref();
   },
-  spawn(command, args, timeout) {
-    log.debug("executing " + command + " with " + args);
+  spawn(command, args, timeout, tries) {
+    if (tries === undefined) {tries = 1;}
+    log.debug("try " + tries + " executing " + command + " with " + args);
```
Comment (ahmedalsudani) [inline platform.js]: @tejohnso if args is a string, it might be worthwhile to do something like `args.join(" ")`

**Comment by fjvallarino** [comment]
:+1:

File: platform.js [hunk 2]
```diff
@@ -121,12 +122,18 @@ module.exports = {
         handled = true;
       });
       child.on("error", (err)=>{
+        log.debug("spawn error " + require("util").inspect(err));
```
Comment (ahmedalsudani) [inline platform.js]: @tejohnso we should log errors to BigQuery as well

File: platform.js [hunk 3]
```diff
@@ -121,12 +122,18 @@ module.exports = {
...
         if (!handled) {rej(err);}
         handled = true;
       });
 
       setTimeout(()=>{
-        if(!handled) {rej(`spawn timeout: ${command} ${args}`);}
+        if(!handled) {
+          log.debug(`try ${tries} spawn timeout: ${command} ${args}`);
+          tries =- 1;
```
Comment (ahmedalsudani) [inline platform.js]: @tejohnso is this supposed to be tries -= 1?

**Comment by tejohnso** [comment]
@ahmedalsudani fixed typo.

**Comment by ahmedalsudani** [comment]
:+1:

**Pull request merged**